### DISCUSSION TITLE
qutebrowser stopped previewing pdfs inline? (for pages that support it)

### DISCUSSION THREAD

**Discussion by xalexalex** [qutebrowser/qutebrowser#6820] [Q&A] [answered]
Hey,
I am pretty sure that qutebrowser did preview pdfs inline, right in the page, for certain pages. Firefox and Chromium do this.
Unfortunately, this has changed about a month ago, and now in place of the inline pdf I get either nothing or a PDF icon with the file name and I have to click to open it.

Example: [link](https://pdfobject.com/examples/replace-PDF-with-second-PDF.html).

Is there something I'm missing?
Thanks

**Comment by toofar** [comment]
Can you run `:version` and see what version of pdf.js it reports? (Or just paste the whole output!)

**Reply by xalexalex** [reply]
pdf.js: 2.11.338 (/usr/share/pdf.js/build/pdf.js)
(I'm on Arch (btw))

**Comment by The-Compiler** [comment, marked as answer]
Embedded PDFs would be rendered by PDFium (Chromium's PDF viewer), but this has been disabled for more than 2 years now: 68b0be563d32965940f4fe4128f8546fa40168bc - see #4911.

**Reply by xalexalex** [reply]
So pdf.js can't render them inline, while PDFium could but is currently buggy and thus disabled?

Thanks!

**Reply by The-Compiler** [reply]
Pretty much. PDFium was buggy in mid-2019. Probably it's much better nowadays. But I don't want to enable it by default because some people prefer opening PDFs in a native PDF viewer. Instead, the `content.pdfjs` setting should probably be renamed/changed to decide between no PDF viewer, PDF.js and PDFium. But so far, nobody picked this up.
